apps/finance/app/src/components/DateRange/DateRangeInput.js

```diff
@@ -92,27 +86,69 @@ class DateRangeInput extends React.PureComponent {
     }
   }
 
+  handleApply = e => {
+    e.preventDefault()
+    e.stopPropagation()
+    this.setState({ showPicker: false }, () => {
+      const { startDate, endDate } = this.state
+      if (startDate && endDate) {
+        this.props.onChange({
+          start: startOfDay(startDate),
+          end: endOfDay(endDate),
+        })
+      }
+    })
+  }
+
+  handleClear = e => {
+    e.preventDefault()
+    e.stopPropagation()
+    this.setState({ showPicker: false, startDate: null, endDate: null }, () =>
+      this.props.onChange({
+        start: null,
+        end: null,
+      })
+    )
+  }
+
   render() {
-    const { startDate, endDate } = this.state
+    const {
+      startDate,
+      endDate,
+      startDateSelected,
+      endDateSelected,
+      showPicker,
+    } = this.state
+    const {
+      compactMode,
+      format,
+      startDate: startDateProps,
+      endDate: endDateProps,
+    } = this.props
 
     const icon = this.state.showPicker ? (
       <IconCalendarSelected />
     ) : (
       <IconCalendar />
     )
-    const value =
-      this.formattedStartDate && this.formattedEndDate
-        ? `${this.formattedStartDate} - ${this.formattedEndDate}`
+    // closed: shows props, if props null then placeholder
+    // opened:
+    //  compact: shows constants, till dates selected
+    //  not compact: shows props, changes with selection
+    const value = !showPicker
```

Comment:
What would you think about making this a function using early returns?

Reply:
Let's make things for readable 😄 !